src/main/kotlin/no/nav/familie/ef/sak/api/beregning/BeregningService.kt

```diff
@@ -1,17 +1,133 @@
+
+            Beløpsperiode(fraOgMedDato = it.fraOgMedDato,
+                          tilDato = it.tilDato,
+                          beløp = beløpTilUtbetalning,
+                          beregningsgrunnlag = Beregningsgrunnlag(samordningsfradrag = samordningsfradrag,
+                                                                  avkortningPerMåned = avkortningPerMåned,
+                                                                  fullOvergangsStønadPerMåned = fullOvergangsStønadPerMåned,
+                                                                  inntekt = inntekt,
+                                                                  grunnbeløp = it.beløp))
+        }
+    }
+
+    private fun beregnAvkortning(grunnbeløp: BigDecimal, inntekt: BigDecimal): BigDecimal {
+        val inntektOverHalveGrunnbeløp = inntekt.subtract(grunnbeløp.multiply(BigDecimal(0.5)))
+        return if (inntektOverHalveGrunnbeløp > BigDecimal(0))
+            inntektOverHalveGrunnbeløp.multiply(BigDecimal(0.45)).setScale(5, RoundingMode.HALF_DOWN) else BigDecimal(0)
+    }
+
+
+    private fun finnStartDatoOgSluttDatoForBeløpsperiode(beløpForInnteksperioder: List<Beløpsperiode>,
+                                                         vedtaksperiodeFraOgmedDato: LocalDate,
+                                                         vedtaksperiodeTilDato: LocalDate): List<Beløpsperiode> {
+        return beløpForInnteksperioder.mapNotNull {
+            if (it.fraOgMedDato.isEqualOrAfter(vedtaksperiodeFraOgmedDato) && it.tilDato.isEqualOrBefore(vedtaksperiodeTilDato)) {
+                it
+            } else if (it.starterFørVedtaksperiodeOgOverlapper(vedtaksperiodeFraOgmedDato, vedtaksperiodeTilDato)) {
+                it.copy(fraOgMedDato = vedtaksperiodeFraOgmedDato)
+            } else if (it.starterEtterVedtaksperiodeOgOverlapper(
+                            vedtaksperiodeFraOgmedDato,
+                            vedtaksperiodeTilDato)) {
+                it.copy(tilDato = vedtaksperiodeTilDato)
+            } else if (it.starterFørOgSlutterEtterVedtaksperiode(vedtaksperiodeFraOgmedDato,
+                                                                 vedtaksperiodeTilDato)) {
+                it.copy(tilDato = vedtaksperiodeTilDato, fraOgMedDato = vedtaksperiodeFraOgmedDato)
+            } else {
+                null
+            }
+        }
+    }
+
+
+    private fun validerVedtaksperioder(vedtaksperiode: List<Periode>) {
+        val sorterteVedtaksperioder = vedtaksperiode.sortedBy { it.fradato }
+
+        sorterteVedtaksperioder.forEachIndexed { index, periode ->
+            if (sorterteVedtaksperioder.size > 1 && index < sorterteVedtaksperioder.lastIndex) {
+                feilHvis(periode.tildato.isEqualOrAfter(sorterteVedtaksperioder[index + 1].fradato)) {
+                    "Vedtaksperioder ${vedtaksperiode} overlapper"
+                }
+            }
+        }
```

Comment:
Samme som den tredje i validerInnteksperioder, dere får vurdere hva dere synes :) 
```suggestion
        feilHvis(sorterteVedtaksperioder.zipWithNext { a, b -> a.tildato.isEqualOrAfter(b.fradato) }.any { !it }) {
            "Vedtaksperioder $vedtaksperiode overlapper"
        }
```

Reply:
enig , fikset her  [30a9237](https://github.com/navikt/familie-ef-sak/pull/630/commits/30a92379601c378b835a2d8cbd2ce634f66a48c4)